gramps_webapi/api/resources/util.py

```diff
@@ -716,3 +724,40 @@ def get_rating(db_handle: DbReadBase, obj: GrampsObject) -> Tuple[int, int]:
                 if citation.confidence > confidence:
                     confidence = citation.confidence
     return count, confidence
+
+
+def add_object(
+    db_handle: DbWriteBase,
+    obj: GrampsObject,
+    trans: DbTxn,
+    fail_if_exists: bool = False,
+):
+    """Commit a Gramps object to the database."""
+    obj_class = obj.__class__.__name__.lower()
+    if fail_if_exists:
+        has_handle = db_handle.method("has_%s_handle", obj_class)
+        if obj.handle and has_handle(obj.handle):
+            raise ValueError("Handle already exists.")
+        has_grampsid = db_handle.method("has_%s_gramps_id", obj_class)
+        if hasattr(obj, "gramps_id") and obj.gramps_id and has_grampsid(obj.gramps_id):
+            raise ValueError("Gramps ID already exists.")
+    try:
+        add_method = db_handle.method("add_%s", obj_class)
+        return add_method(obj, trans)
+    except AttributeError:
+        raise ValueError("Database does not support writing.")
+    raise ValueError("Unexpected object type.")
```

Comment:
Isn't this line dead?

It looks like we either already returned, or some error is already raised at this point.